proposer/proposer.go

```diff
@@ -166,6 +168,12 @@ func (p *Proposer) ProposeOp(ctx context.Context) error {
 		return fmt.Errorf("failed to wait until L2 execution engine synced: %w", err)
 	}
 
+	log.Info("Comparing proposer TKO balance to block fee")
+
+	if err := p.checkTaikoTokenBalance(); err != nil {
```

Comment:
hmmm check the balance when initing or before each proposing operation, not sure which approach is better?

Reply:
i moved it to ProposeTxList, is that better?

Reply:
will move it to `ProposeOp` better? if the proposer doesn't have enough balance, we won't waste a RPC call to fetch the mempool?

Reply:
OK!